docs/source/user_guide.rst

```diff
@@ -0,0 +1,184 @@
+User Guide
+==========
+
```

Comment:
```suggestion
DeepInverse provides `PyTorch <https://pytorch.org/>_` tools for solving inverse problems.
```

I think it's important at this stage users understand we are using torch

Reply:
I think we are already saying this in the home page, it might be a bit repetitive